src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
-        }
+            private static byte[] BuildRequestHeader(Uri uri, ClientWebSocketOptions options, string secKey)
+            {
+                StringBuilder builder = t_cachedStringBuilder ?? (t_cachedStringBuilder = new StringBuilder());
+                Debug.Assert(builder.Length == 0, $"Expected builder to be empty, got one of length {builder.Length}");
+                try
+                {
+                    builder.Append("GET ").Append(uri.PathAndQuery).Append(" HTTP/1.1\r\n");
 
-        public Task CloseAsync(
-            WebSocketCloseStatus closeStatus,
-            string statusDescription,
-            CancellationToken cancellationToken)
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the required headers
+                    builder.Append("Host: ").Append(uri.Host).Append("\r\n");
+                    builder.Append("Connection: Upgrade\r\n");
+                    builder.Append("Upgrade: websocket\r\n");
+                    builder.Append("Sec-WebSocket-Version: 13\r\n");
+                    builder.Append("Sec-WebSocket-Key: ").Append(secKey).Append("\r\n");
 
-        public Task CloseOutputAsync(
-            WebSocketCloseStatus closeStatus,
-            string statusDescription,
-            CancellationToken cancellationToken)
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the additionally requested headers
+                    foreach (string key in options.RequestHeaders.AllKeys)
+                    {
+                        builder.Append(key).Append(": ").Append(options.RequestHeaders[key]).Append("\r\n");
+                    }
 
-        public void Dispose()
-        {
-        }
+                    // Add the optional subprotocols header
+                    if (options.RequestedSubProtocols.Count > 0)
+                    {
+                        builder.Append(HttpKnownHeaderNames.SecWebSocketProtocol).Append(": ");
+                        builder.Append(options.RequestedSubProtocols[0]);
+                        for (int i = 1; i < options.RequestedSubProtocols.Count; i++)
+                        {
+                            builder.Append(", ").Append(options.RequestedSubProtocols[i]);
+                        }
+                        builder.Append("\r\n");
+                    }
 
-        public void Abort()
-        {
+                    // Add an optional cookies header
+                    if (options.Cookies != null)
+                    {
+                        string header = options.Cookies.GetCookieHeader(uri);
+                        if (!string.IsNullOrWhiteSpace(header))
+                        {
+                            builder.Append(HttpKnownHeaderNames.SetCookie).Append(": ").Append(header).Append("\r\n");
```

Comment:
I looked in Wireshark at what happens when super long cookies are sent with ClientWebSocket in both the desktop and with corefx (e.g. 5 cookies each 4096 bytes long), and I don't see any such thing being done... there's just a single Set-Cookie header with all of the cookies as part of it.  My understanding of header limits is typically, if a server does impose one even though HTTP doesn't have an explicit limit, that the limit typically applies to all headers, not to a single header, so chopping into multiple Set-Cookie headers wouldn't help.